### DISCUSSION TITLE
Enforcing fixed (initial) head values?

### DISCUSSION THREAD

**Discussion by nelerey** [pastas/pastas#761] [Q&A] [answered]
_This is a general question about pastas functionality._

For some applications (e.g. gap-filling or forecasting), we want to simulate time series where fixed initial conditions (or fixed head values at arbitrary time steps) are provided by observations. In such cases, we'd like the simulations to match the observations at time steps where observations are available. For time steps/periods where we only have stress/forcing data and no head observations, we'd also like to estimate the uncertainty of the simulations, which widens from zero as we move further from at time steps where we have observations (assuming perfectly reliable observations for simplicity).

Is this currently possible in pastas, and if so, what is the recommended way to approach this?

**Comment by dbrakenhoff** [comment, marked as answer]
Hi @nelerey, thanks for your question. Pastas currently does not have that functionality, but the package [metran](https://github.com/pastas/metran)  can do something along those lines using a Kalman smoother. Metran uses information from other similar time series (e.g. other head observations) to simulate heads and can compute uncertainties that increase as you get further away from the previous measurement. Check out the notebooks there to see an example of that. 

I don't know enough about how that all works to be able to say if this could also be applied to Pastas. It probably can(?), but that hasn't been done up till now.
